pycaret/internal/pycaret_experiment/time_series_experiment.py

```diff
@@ -3265,3 +3272,361 @@ def get_fold_generator(
                     start_with_window=True,
                 )
         return fold_generator
+
+    def test_model(
+        self,
+        estimator: Optional[Any] = None,
+        test: str = "all",
+        alpha: float = 0.05,
+        split: str = "all",
+    ) -> pd.DataFrame:
+        if estimator is None:
+            data = self._get_y_data(split=split)
+            results = test_(data=data, test=test, alpha=alpha)
+        else:
+            raise NotImplementedError(
+                "Tests on estimators have not been implemented yet."
+            )
+        return results
+
+    def _get_y_data(self, split="all"):
+        if split == "all":
+            data = self.y.copy()
+        elif split == "train":
+            data = self.y_train.copy()
+        else:
+            raise ValueError("split value: '{split}' is not supported.")
+        return data
+
+
+#############################################################
+###################### TESTS STARTS HERE ####################
+#############################################################
+
+#################
+#### Helpers ####
+#################
+
+
+def test_(data, test: str, alpha: float = 0.05, *kwargs):
+    if test == "all":
+        results = test_all(data=data, alpha=alpha)
+    elif test == "stat_summary":
+        results = test_summary_statistics(data=data)
+    elif test == "white_noise":
+        results = test_is_white_noise(data=data, alpha=alpha, verbose=True)[1]
+    elif test == "stationarity":
+        results = test_is_stationary(data=data, alpha=alpha)
+    elif test == "adf":
+        results = test_is_stationary_adf(data=data, alpha=alpha, verbose=True)[1]
+    elif test == "kpss":
+        results = test_is_stationary_kpss(data=data, alpha=alpha, verbose=True)[1]
+    elif test == "normality":
+        results = test_is_gaussian(data=data, alpha=alpha, verbose=True)[1]
+    else:
+        raise ValueError(f"Tests: '{test}' is not supported.")
+    return results
```

Comment:
perhaps we should use a dict here. This is fine too, though

Reply:
No, a dict of `{"test_name": test_func}` instead of ifs. But it's not really a huge deal, and it will need a little effort.